### PR TITLE
Delete scripts

### PR DESCRIPTION
**Pull request by squidsoup**
## Note
This is dependent on [#183](https://github.com/canonical-web-and-design/maas-ui/pull/183). Leaving this on WIP until 183's merged, to make this easier to review.

## Done
* Implement deleting scripts.

## QA
* Visit testing/commissioning scripts.
* Ensure default scripts (those provided by MAAS) cannot be deleted.
* Upload a testing or commissioning script via old maas, revisit the testing/commissioning page in maas-ui to update the list.
* Delete the script, you should see a confirmation and the script should disappear from the list.

### CODE REVIEW

File: ui/src/app/settings/reducers/scripts/scripts.js [hunk 1]
```diff
@@ -16,6 +16,22 @@ const scripts = produce(
         draft.loaded = false;
         draft.error = action.error;
         break;
+      case "DELETE_SCRIPT_START":
```
Comment (huwshimi) [inline ui/src/app/settings/reducers/scripts/scripts.js]: Do we also need to handle `DELETE_SCRIPT_ERROR` here?

File: ui/src/app/settings/views/Scripts/ScriptsList.js [hunk 2]
```diff
@@ -118,6 +127,8 @@ const generateRows = (
             onCancel={hideExpanded}
             onConfirm={() => {
               // TODO: actually delete
```
Comment (huwshimi) [inline ui/src/app/settings/views/Scripts/ScriptsList.js]: I think it now actually does!

File: ui/src/app/settings/views/Scripts/ScriptsList.js [hunk 3]
```diff
@@ -157,6 +170,16 @@ const ScriptsList = ({ type = "commissioning" }) => {
     dispatch(actions.scripts.fetch());
   }, [dispatch, type]);
 
+  useEffect(() => {
```
Comment (huwshimi) [inline ui/src/app/settings/views/Scripts/ScriptsList.js]: I guess we also need one of these for handling error messages.

**Comment by squidsoup** [comment]
Now displays error e.g. 
image

**Comment by huwshimi** [comment]
Strangely this is what I now get when deleting a script:

Screen Shot 2019-09-10 at 2 41 13 pm

**Comment by lyubomir-popov** [comment]
After deleting the scripts, we see 0:
image
Can that turn to a #666 no scripts available message? @lilyvidenova I think we made a design for this somewhere recently?

**Comment by lyubomir-popov** [comment]
Would it be too much to patch the side nav on small screens as a drive by?
All it would take is:

```
@media (max-width: $breakpoint-medium) {
  .settings-nav__list {
      list-style: none;
      columns: 2;
      margin-left: 0;
      padding-left: 0;
      margin-bottom: 0.5rem;
      margin-top: -0.25rem;
  }
  .settings-nav__list .settings-nav__list .settings-nav__item + .settings-nav__item {
    border-top: 1px solid #cdcdcd;
  }
}
```
image

**Comment by lilyvidenova** [comment]
Hmm I'm getting lots of error notifications:

Screenshot 2019-09-10 at 11 56 39

**Comment by squidsoup** [comment]
> After deleting the scripts, we see 0:
> image
> Can that turn to a #666 no scripts available message? @lilyvidenova I think we made a design for this somewhere recently?

We have a card for this one https://github.com/canonical-web-and-design/maas-ui/issues/184

**Comment by squidsoup** [comment]
> Hmm I'm getting lots of error notifications:
> 
> Screenshot 2019-09-10 at 11 56 39

@lilyvidenova that's not an error I've seen, are you able to provide steps to reproduce that?

**Pull request merged**